src/Kongverge/Services/KongAdminHttpClient.cs

```diff
@@ -9,7 +11,13 @@ public class KongAdminHttpClient : HttpClient
     {
         public KongAdminHttpClient(IOptions<Settings> configuration) : base(new EnsureSuccessHandler())
         {
-            BaseAddress = new Uri($"http://{configuration.Value.Admin.Host}:{configuration.Value.Admin.Port}");
+            var adminConfig = configuration.Value.Admin;
+
+            BaseAddress = new Uri($"http://{adminConfig.Host}:{adminConfig.Port}");
+
+            if (string.IsNullOrEmpty(adminConfig.AuthUsername)) return;
```

Comment:
The code style in this repository is to always have braces for if-statements (not my choice).